packages/kbn-grid-layout/grid/grid_panel.tsx

```diff
@@ -129,6 +131,63 @@ export const GridPanel = forwardRef<
       []
     );
 
+    useEffect(
+      () => {
+        /** Update the styles of the panel via a subscription to prevent re-renders */
+        const styleSubscription = combineLatest([
+          gridLayoutStateManager.gridLayout$,
+          gridLayoutStateManager.expandedPanelId$,
+          gridLayoutStateManager.isMobileView$,
+        ])
+          .pipe(skip(1)) // skip the first emit because the `initialStyles` will take care of it
+          .subscribe(([gridLayout, expandedPanelId, isMobileView]) => {
+            const ref = gridLayoutStateManager.panelRefs.current[rowIndex][panelId];
+            const allPanels = gridLayout[rowIndex].panels;
+            const panel = allPanels[panelId];
+            if (!ref || !panel) return;
+
+            if (expandedPanelId && expandedPanelId === panelId) {
+              // Translate the expanded panel back to its initial position
+              // since all other GridRow elements have been moved off-screen
+              ref.style.transform = 'translate(9999px, 9999px)';
+
+              // Stretch the expanded panel to occupy the remaining available space in the viewport.
+              ref.style.position = `absolute`;
+              ref.style.top = `0`;
+              ref.style.left = `0`;
+              ref.style.width = `100%`;
+              ref.style.height = `100%`;
+              return;
+            } else {
+              ref.style.position = ``;
+              ref.style.transform = ``;
```

Comment:
Hmmm - this is causing the panel to "glitch" as I drag it around:

https://github.com/user-attachments/assets/9d796be7-b40c-459f-b1dc-03fa5c357c03

I think we need to be more descriptive about when to set these styles - i.e. I think we should also only set them if
`expandedPanelId === panelId`, but only when there is no expanded panel iD

Reply:
My above suggestion doesn't work - instead, we should remove `gridLayoutStateManager.gridLayout$` from the `combineLatest` and use `gridLayoutStateManager.gridLayout$.getValue()` in the subscription instead.

Anything in the `combineLatest` causes the subscription to fire whenever it emits - but do we only need to set the expanded panel styles when `expandedPanelId` emits, not when `gridLayoutStateManager.gridLayout$` emits

Reply:
https://github.com/elastic/kibana/pull/200771/commits/53ee088facd4e2ca5fbb922f9b7e1bd697e9d4aa